### ISSUE TITLE
Increase generated cert key size to at least 3072 bits

### ISSUE THREAD

**Issue by janmedrek** [kyma-project/lifecycle-manager#1771]
### Description

All certificate key lengths should be at least 3072 bits if RSA is used. This applies to all the certificates generated/used by both Lifecycle Manager and Runtime Watcher.

### Reasons

We want to fulfil security requirements and recommendations.

### Acceptance Criteria

- [x] Increase the key size for the generated certificates to 4096
- [x] Keep using the RSA algorithm
- [ ] Check if there will be some downtime, and if it resolves by itself
    - [ ] Communicate this information to the SREs
- [x] Create a follow-up ticket to investigate the possibility of using the ECC algorithm
  - Does it change anything on the client side?

### Attachments

We can ignore the Runtime Watcher downtime as long as it does not affect the KLM behaviour - we have 5 minutes interval so the impact would be minimal for affected clusters.

API ref: https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.PrivateKeyAlgorithm

Code changes:
- https://github.com/kyma-project/lifecycle-manager/blob/a180040ef2e6b8742a61ccc44eaaabae1dded8f5/config/watcher/certificate_setup.yaml#L27
- https://github.com/kyma-project/lifecycle-manager/blob/49a4feeb102523184b1945dc3147c008a9299e3a/pkg/watcher/certificate_manager.go#L208

**Comment by jeremyharisch** [comment]
Places which need to be changed:
- https://github.com/kyma-project/lifecycle-manager/blob/a180040ef2e6b8742a61ccc44eaaabae1dded8f5/config/watcher/certificate_setup.yaml#L27
- https://github.com/kyma-project/lifecycle-manager/blob/49a4feeb102523184b1945dc3147c008a9299e3a/pkg/watcher/certificate_manager.go#L208
- Maybe the Issuers need to be updated

Please keep in mind, that we might have a downtime of the watcher

**Comment by jeremyharisch** [comment]
Questions need to clarified:
- Is Watcher communication relevant for module requirements?
- If yes:
    - Which algorithm to use? Symentric/Asymetric?
    - What is min length? Since 3072 is invalid for RSA for private key length

Internal issues: 
- kyma/backlog/issues/5825
- kyma/backlog/issues/6006#issuecomment-7333169

**Comment by amritanshusikdar** [comment]
For the algorithm part, the new industry standard has chosen to drop off RSA and have **ED25519** as the new default. It is better than RSA in numerous ways, however the only catch is RSA is also fully compatible with legacy systems, unlike ED25519 which has support from OpenSSH version 6.5 and later. Considering our use-case it can be assumed that almost all the machines involved in the business are up-to-date with OpenSSH v6.5, as the older versions use MD5 signatures, which are anyway not secure, so ED25519 is a good solution to the algorithm upgrade.

Following are some resources suggesting adoption of ED25519 as the new industry standard:
1. [GitHub documentation for adding new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
2. [More GitHub documentations](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
3. [GitLab documentation](https://docs.gitlab.com/ee/user/ssh.html). Search for the terms: `ED25519 is more secure than RSA`
4. [Latest commit of OpenSSH as of 19.08.2024 declaring ED25519 as the new default](https://github.com/openssh/openssh-portable/commit/a0b35c791cad1f85481b23ba46373060292e1c80)
5. [Linux man-pages](https://man7.org/linux/man-pages/man1/ssh-keygen.1.html). Search for the terms: `If invoked without any arguments`
6. Last but not the least, [something about the shorter ED25519 key length](https://security.stackexchange.com/questions/101044/is-it-bad-that-my-ed25519-key-is-so-short-compared-to-a-rsa-key)

**Comment by janmedrek** [comment]
> Is Watcher communication relevant for module requirements?
Yes, we need to address all the communication "channels" we use, this requirement is common for all the components, not only modules.

> Which algorithm to use? Symmetric/Asymmetric?
The recommendation was to go with RSA or ECC (Elliptical Curves), which both are asymmetric. AFAIK, the ECC performs better when generating keys, we are free to choose though.

> What is min length? Since 3072 is invalid for RSA for private key length
Anything above 3072 is good, let's be reasonable there and not go to the extremes. I'd say the first "step" above that mark is fine, so that would be at least 4096 for RSA and 256 for ECC.

**Comment by c-pius** [comment]
Implemented via https://github.com/kyma-project/lifecycle-manager/pull/1794

Locally tested and verified that the credentials are rotated within a couple of seconds. To do this, we had to set [`rotationPolicy`](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificatePrivateKey) to `Always`. This is also the recommended value by certmanager: https://cert-manager.io/docs/usage/certificate/#the-rotationpolicy-setting. The rotation will also need to be verified when promoting through dev and stage. 

There were issues when testing the same for 8192 bit key sizes. Hence, setting this is prevented for now and a follow up to investigate was created if we want to stick with RSA algorithm: https://github.com/kyma-project/lifecycle-manager/issues/1793
